src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1194,12 +1194,17 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			if (this.useConnectionString) {
-				const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
-				if (connInfo) {
+				try {
```

Comment:
why are we adding a try-catch block here? does buildConnectionInfo method throw an error?

Reply:
Yes, that was another issue I tried to address here: https://github.com/Azure/azure-cosmosdb-ads-extension/pull/55/files, an extension should be able to do a Promise.reject but that was not handled and was just hanging here. This change basically hardens it also for the case an extension has a bug.

Reply:
Makes sense, we should generally always be assuming that any calls to external providers can throw and making sure we handle that appropriately.